lib/luasocket.c

```diff
@@ -63,8 +63,10 @@ static void luasocket_checkaddr(lua_State *L, struct socket *socket, luasocket_a
 		struct sockaddr_in *addr_in = (struct sockaddr_in *)addr;
 		addr_in->sin_addr.s_addr = htonl((u32)luaL_checkinteger(L, ix));
 		addr_in->sin_port = htons((u16)luaL_checkinteger(L, ix + 1));
-	}
-	else {
+	} else if (addr->family == AF_PACKET) {
+		struct sockaddr_ll *addr_ll = (struct sockaddr_ll *)addr;
+		addr_ll->sll_ifindex = (int)luaL_checkinteger(L, ix);
+	} else {
```

Comment:
```suggestion
	}
	else if (addr->family == AF_PACKET) {
		struct sockaddr_ll *addr_ll = (struct sockaddr_ll *)addr;
		addr_ll->sll_ifindex = (int)luaL_checkinteger(L, ix);
	}
	else {
```
I would do that for both preserving the code style and to avoid change unnecessary lines in the patch.. I don't mind to change the code style, but I would do this as a separate commit and change the code base as a whole.. at least the whole module..

Reply:
that was my bad. I didn't notice the style. I always go with whatever is already there.

Reply:
It's actually my bad it's not documented 😉